### DISCUSSION TITLE
`mise` fails to install `kubectl` on Windows from Aqua registry

### DISCUSSION THREAD

**Discussion by dangmai** [jdx/mise#3885] [Q&A]
Using this `mise.toml`:

```toml
[tools]
kubectl = '1.32.0'
```

`mise` will fail to install `kubectl` on Windows with the following message:

```
mise ERROR failed to install aqua:kubernetes/kubectl@1.32.0
mise ERROR HTTP status client error (404 Not Found) for url (https://cdn.dl.k8s.io/1.32.0/bin/windows/amd64/kubectl)        tl)                                                                                                                4/kubectl)   
mise ERROR HTTP status client error (404 Not Found) for url (https://cdn.dl.k8s.io/release/v1.32.0/bin/windows/amd6d64/kubectl)
mise ERROR Run with --verbose or MISE_VERBOSE=1 for more information
```

This is because `aqua` automatically adds the `.exe` extension on Windows platform, while `mise` does not. There was a PR on Aqua's repository to make the `.exe` extension explicit in the URL, but the maintainer has closed it: https://github.com/aquaproj/aqua-registry/pull/30398 (with a follow up note for `mise` users [here](https://github.com/aquaproj/aqua-registry/issues/30430)).

I think `mise` logic should follow `aqua` logic when determining the URL to use while downloading dependencies.

**Comment by jdx** [comment]
it'll be awhile until I get back home where my windows laptop is to be able to do this.

I believe the logic would need to be added somewhere around here though: https://github.com/jdx/mise/blob/cc88dca50e8e0dac94dbb83d0ce1ebcfc38a1ec4/src/backend/aqua.rs#L203

**Comment by roele** [comment]
It seems my [fix](https://github.com/jdx/mise/pull/4006) might not be enough as the `AquaPackage.url` also needs the `.exe` extension (https://cdn.dl.k8s.io/release/v1.32.0/bin/windows/amd64/kubectl.exe). What bugs me is the [registry](https://github.com/aquaproj/aqua-registry/blob/main/pkgs/kubernetes/kubectl/registry.yaml) that contains an override for the checksum url using an explicit pattern with `.exe` but there is none for the binary. I have the feeling that a proper fix should rather have been to provide a url template for windows.

**Reply by jdx** [reply]
yeah I agree. I don't like the hidden behaviors like this that aqua sometimes uses. That said, I don't think he ever intended people to use just the registry from aqua so it's not like we're a use-case that was designed for.
